x-pack/metricbeat/module/aws/s3_daily_storage/s3_daily_storage.go

```diff
@@ -205,13 +206,21 @@ func createCloudWatchEvents(outputs []cloudwatch.MetricDataResult, regionName st
 	// AWS s3_daily_storage metrics
 	mapOfMetricSetFieldResults := make(map[string]interface{})
 	// Find a timestamp for all metrics in output
-	if len(outputs) > 0 && len(outputs[0].Timestamps) > 0 {
+	timestamp := time.Time{}
+	for _, output := range outputs {
+		if output.Timestamps != nil {
+			timestamp = output.Timestamps[0]
```

Comment:
Should we also check if len(output.Timestamp) > 0 ? 

Could also add a note to the code on why we just take the first timestamp we find? Are all timestamps the same or how do they differ?

Reply:
This is actually a common issue in all aws module code. For example, getMetricData results can look like:
```
[...
{
  Id: "ec210",
  Label: "i-0c68eeb552231a8d0 NetworkOut",
  StatusCode: Complete,
  Timestamps: [2019-03-11 18:37:00 +0000 UTC],
  Values: [6938.6]
} {
  Id: "ec28",
  Label: "i-0c68eeb552231a8d0 CPUCreditUsage",
  StatusCode: Complete,
  Timestamps: [2019-03-11 18:42:00 +0000 UTC,2019-03-11 18:37:00 +0000 UTC],
  Values: [0.008226,0.006062]
}
...]
```
This case, we will need to find the common timestamp and use that to collect values from each metric. I will remove the changes here and add it in a separate PR based on https://github.com/elastic/beats/pull/11142.